lib/workers/global/ssh_socket.ts

```diff
@@ -0,0 +1,40 @@
+import type { HostRule } from '../../types';
+import * as cp from 'child_process';
+import * as net from 'net';
+import Agent from 'ssh-agent-js/client/index';
+import { join } from 'upath';
+
+export class SshSocket {
+  agent_socket: string;
+  agent_process: cp.ChildProcess;
+  socket: net.Socket;
+  agent: Agent;
+
+  async start(cacheDir: string) {
+    this.agent_socket = join(
+      cacheDir,
+      '.ssh-agent.' + Math.floor(Math.random() * 999999) + 1
+    );
+    this.agent_process = cp.spawn('ssh-agent', ['-d', '-a', this.agent_socket]);
+    await new Promise((r) => setTimeout(r, 2000));
+    this.socket = net.connect(this.agent_socket);
```

Comment:
do we need to store a reference to socket.

Reply:
I thought we do, but we don't. The error was somewhere else